src/ra_log_wal.erl

```diff
@@ -714,46 +740,80 @@ dump_records(<<_:1/unsigned, 1:1/unsigned, _:22/unsigned,
 dump_records(<<>>, Entries) ->
     Entries.
 
-try_recover_records(Data, Cache) ->
-    try recover_records(Data, Cache) of
-        ok -> ok
-    catch _:_ = Err ->
-              ?WARN("wal: encountered error during recovery: ~w~n"
-                    "Continuing.~n", [Err]),
-              ok
-    end.
+recover_wal_chunks(Fd, RecoveryChunkSize, Cache) ->
+    Chunk = read_from_wal_file(Fd, RecoveryChunkSize),
+    recover_records(Fd, Chunk, Cache, RecoveryChunkSize).
 
-recover_records(<<Trunc:1/unsigned, 0:1/unsigned, IdRef:22/unsigned,
+% Reached zeros indicating end of wal file
+recover_records(_, <<_:1/unsigned, 0:1/unsigned, _:22/unsigned, 
+                    IdDataLen:16/unsigned, _:IdDataLen/binary,
+                    _:32/integer, 0:32/unsigned, _/binary>>,
+                  _, _) ->
+    ok;
+% First record or different UID to last record
+recover_records(Fd, <<Trunc:1/unsigned, 0:1/unsigned, IdRef:22/unsigned,
                   IdDataLen:16/unsigned, UId:IdDataLen/binary,
                   Checksum:32/integer,
                   EntryDataLen:32/unsigned,
                   Idx:64/unsigned, Term:64/unsigned,
                   EntryData:EntryDataLen/binary,
-                  Rest/binary>>, Cache) ->
-    % first writer appearance in WAL
+                  Rest/binary>>,
+                Cache, RecoveryChunkSize) ->
     true = validate_and_update(UId, Checksum, Idx, Term, EntryData, Trunc),
-    % TODO: recover writers info, i.e. last index seen
-    recover_records(Rest,
-                    Cache#{IdRef =>
-                           {UId, <<1:1/unsigned, IdRef:22/unsigned>>}});
-recover_records(<<Trunc:1/unsigned, 1:1/unsigned, IdRef:22/unsigned,
+    recover_records(Fd, Rest,
+                    Cache#{
```

Comment:
This kind of map formatting isn't done anywhere else in this module so looks a bit out of place here.

Reply:
Might be nicer to assign a new Cache variable on the line before